pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -325,6 +326,22 @@ func (w *Webhook) inject(pod *corev1.Pod, _ string, _ dynamic.Interface) error {
 	return w.injectAutoInstruConfig(pod, libsToInject, autoDetected, injectionType)
 }
 
+func injectSecurityClientLibraryProducts(pod *corev1.Pod) {
+	injectProductActivationKey(pod, "asm.enabled", "DD_APPSEC_ENABLED")
+	injectProductActivationKey(pod, "iast.enabled", "DD_IAST_ENABLED")
+	injectProductActivationKey(pod, "sca.enabled", "DD_SCA_ENABLED")
+}
+
+func injectProductActivationKey(pod *corev1.Pod, configKey string, envVarKey string) {
+	if (config.Datadog.Get(configKey) != nil) {
```

Comment:
if `BindEnvAndSetDefault` was used instead of `BindEnv` (as suggested at the end of the PR), you wouldn't need to do this extra `nil` check.